client/client-http/src/main/java/software/amazon/smithy/java/client/http/plugins/RequestCompressionPlugin.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package software.amazon.smithy.java.client.http.plugins;
+
+import java.util.List;
+import software.amazon.smithy.java.client.core.AutoClientPlugin;
+import software.amazon.smithy.java.client.core.ClientConfig;
+import software.amazon.smithy.java.client.core.interceptors.ClientInterceptor;
+import software.amazon.smithy.java.client.core.interceptors.RequestHook;
+import software.amazon.smithy.java.client.http.HttpContext;
+import software.amazon.smithy.java.client.http.HttpMessageExchange;
+import software.amazon.smithy.java.client.http.compression.CompressionAlgorithm;
+import software.amazon.smithy.java.client.http.compression.Gzip;
+import software.amazon.smithy.java.context.Context;
+import software.amazon.smithy.java.core.schema.TraitKey;
+import software.amazon.smithy.java.http.api.HttpRequest;
+import software.amazon.smithy.java.io.datastream.DataStream;
+import software.amazon.smithy.model.traits.RequestCompressionTrait;
+import software.amazon.smithy.utils.ListUtils;
+import software.amazon.smithy.utils.SmithyInternalApi;
+
+/**
+ * Compress the request body using provided compression algorithm if @requestCompression trait is applied.
+ */
+@SmithyInternalApi
+public final class RequestCompressionPlugin implements AutoClientPlugin {
+
+    @Override
+    public void configureClient(ClientConfig.Builder config) {
+        if (config.isUsingMessageExchange(HttpMessageExchange.INSTANCE)) {
+            config.addInterceptor(RequestCompressionInterceptor.INSTANCE);
+            config.putConfigIfAbsent(HttpContext.DISABLE_REQUEST_COMPRESSION, false);
```

Comment:
Why do we need this? How can I disable compression even if the plugin is present?